client/my-sites/earn/refer-a-friend/index.tsx

```diff
@@ -1,39 +1,36 @@
 import { localizeUrl } from '@automattic/i18n-utils';
 import { useTranslate } from 'i18n-calypso';
 import { compact } from 'lodash';
-import { FunctionComponent, Fragment, useState, useEffect } from 'react';
-import { connect } from 'react-redux';
+import { Fragment, useState, useEffect } from 'react';
 import earnSectionImage from 'calypso/assets/images/earn/earn-section.svg';
 import referralImage from 'calypso/assets/images/earn/referral.svg';
 import ClipboardButtonInput from 'calypso/components/clipboard-button-input';
 import PromoSection, { Props as PromoSectionProps } from 'calypso/components/promo-section';
 import { CtaButton } from 'calypso/components/promo-section/promo-card/cta';
 import wp from 'calypso/lib/wp';
+import { useDispatch, useSelector } from 'calypso/state';
 import { bumpStat, composeAnalytics, recordTracksEvent } from 'calypso/state/analytics/actions';
 import isSiteAutomatedTransfer from 'calypso/state/selectors/is-site-automated-transfer';
 import { isJetpackSite } from 'calypso/state/sites/selectors';
-import getSiteBySlug from 'calypso/state/sites/selectors/get-site-by-slug';
-import { IAppState } from 'calypso/state/types';
-import { getSelectedSiteSlug } from 'calypso/state/ui/selectors';
-import type { SiteSlug } from 'calypso/types';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
 
 import './style.scss';
 
-interface ConnectedProps {
-	siteId: number | undefined;
-	selectedSiteSlug: SiteSlug;
-	isJetpack: boolean;
-	isAtomicSite: boolean;
-	trackCtaButton: ( feature: string ) => void;
-}
-
-const ReferAFriendSection: FunctionComponent< ConnectedProps > = ( {
-	isJetpack,
-	isAtomicSite,
-	trackCtaButton,
-} ) => {
+const ReferAFriendSection = () => {
 	const translate = useTranslate();
+	const dispatch = useDispatch();
 	const [ peerReferralLink, setPeerReferralLink ] = useState( '' );
+	const site = useSelector( ( state ) => getSelectedSite( state ) );
+	const isJetpack = useSelector( ( state ) => isJetpackSite( state, site?.ID ) );
+	const isAtomicSite = useSelector( ( state ) => isSiteAutomatedTransfer( state, site?.ID ) );
+
+	const trackCtaButton = ( feature: string ) =>
+		dispatch(
+			composeAnalytics(
+				recordTracksEvent( 'calypso_earn_page_cta_button_click', { feature } ),
+				bumpStat( 'calypso_earn_page', 'cta-button-' + feature )
+			)
+		);
```

Comment:
I think you can simplify this now to this and it'll still work:

```js
import { recordTracksEvent } from '@automattic/calypso-analytics';

const trackCtaButton = ( feature: string ) => {
  recordTracksEvent( 'calypso_earn_page_cta_button_click', { feature } );
  bumpStat( 'calypso_earn_page', 'cta-button-' + feature );
}
```

Reply:
I removed the dispatch. I kept the composeAnalytics() as I'm not 100% sure what the code [here](https://github.com/Automattic/wp-calypso/blob/a1ec9c8d83ac1ae8d8bd805e263ebb1262146a8b/client/state/analytics/actions/compose-analytics.js#L4) is doing and don't want to risk altering the data that's being collected. I did re-test after this change to confirm that the tracks event is still being fired.